airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -606,3 +636,303 @@ def object_exists(self, uuid: str | UUID, **kwargs) -> bool:
+        )
+
+        return df, uuid_column
+
+    def _check_existing_documents(
+        self, data: pd.DataFrame, document_column: str, class_name: str, uuid_column: str
+    ) -> tuple[set, set]:
+        """
+        Get all object uuids belonging to a document.
+
+        :param data: A single pandas DataFrame.
+        :param document_column: The name of the property to query.
+        :param class_name: The name of the class to query.
+        :param uuid_column: The name of the column containing the UUID.
+        """
+        offset = 0
+        limit = 2000
+        documents_to_uuid: dict = {}
+        existing_documents = set()
+        document_keys = set(data[document_column])
+        non_existing_documents = document_keys.copy()
+        while True:
+            data_objects = (
+                self.conn.query.get(properties=[document_column], class_name=class_name)
+                .with_additional([uuid_column])
+                .with_where(
+                    {
+                        "operator": "Or",
+                        "operands": [
+                            {"valueText": key, "path": document_column, "operator": "Equal"}
+                            for key in document_keys
+                        ],
+                    }
+                )
+                .with_offset(offset)
+                .with_limit(limit)
+                .do()["data"]["Get"][class_name]
+            )
+            if len(data_objects) == 0:
+                break
+            for data_object in data_objects:
+                document_url = data_object[document_column]
+
+                if document_url not in documents_to_uuid:
+                    documents_to_uuid[document_url] = set()
+                    existing_documents.add(document_url)
+                    non_existing_documents.remove(document_url)
+
+                documents_to_uuid[document_url].add(data_object["_additional"][uuid_column])
+            offset = offset + limit
+        return existing_documents, non_existing_documents
+
+    def _delete_all_documents_objects(
+        self,
+        document_keys: list[str],
+        document_column: str,
+        class_name: str,
+        batch_delete_error: list | None = None,
+        tenant: str | None = None,
+        batch_config_params: dict[str, Any] | None = None,
+    ):
+        if not batch_config_params:
+            batch_config_params = {}
+
+        # configuration for context manager for __exit__ method to callback on errors for weaviate
+        # batch ingestion.
+        if not batch_config_params.get("callback"):
+            batch_config_params.update({"callback": partial(self.process_batch_errors, batch_delete_error)})
+
+        self.conn.batch.configure(**batch_config_params)
+
+        with self.conn.batch as batch:
+            batch.consistency_level = weaviate.data.replication.ConsistencyLevel.ALL
```

Comment:
We should comment on consistency between tenants. (in terms of why is consistency needed?)

Reply:
Sure, makes sense. I'll add a comment.

Reply:
Added code comment explaining the need for `consistencyLevel.ALL`